iac-extensions/arm/src/main/java/org/sonar/iac/arm/tree/impl/bicep/DecoratorImpl.java

```diff
@@ -50,4 +50,8 @@ public Kind getKind() {
   public Expression expression() {
     return expression;
   }
+
+  public SyntaxToken keyword() {
+    return keyword;
+  }
```

Comment:
Same here. This should be represented in an interface as all the other getters are represented in the interface. How to know from the tree, without casting to the implementation what getter is available. The parent interfaces should even extend a common interface like `hasKeyword` or `hasIdentifier`

Reply:
I thought about introducing interface like `HasKeyword` but I wasn't sure about it. 
I'm ok to introduce it, however, e.g. `OutputDeclaration` should not extend new interface because it is a common interface for Bicep & JSON where keyword doesn't exist in JSON.

Reply:
Fine for me, then let at least the Bicep implementation implement the framework. That way you can use the reference in the highlighting visitor.

Reply:
`HasKeyword` is created